### PR TITLE
Supertokens integration

### PR DESCRIPTION
**Pull request by nkshah2**

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
   "version": "0.0.0",
   "private": true,
   "dependencies": {
-    "@redwoodjs/api": "^0.39.3",
-    "@redwoodjs/graphql-server": "^0.39.3"
+    "@middy/core": "^2.5.3",
```
Comment (rishabhpoddar) [inline api/package.json:6]: We don't need middy

File: api/src/functions/graphql.js [hunk 2]
```diff
@@ -3,11 +3,15 @@ import { createGraphQLHandler } from '@redwoodjs/graphql-server'
 import directives from 'src/directives/**/*.{js,ts}'
 import sdls from 'src/graphql/**/*.sdl.{js,ts}'
 import services from 'src/services/**/*.{js,ts}'
+import { init } from "supertokens-node";
+import { config } from "src/services/supertokens";
```
Comment (rishabhpoddar) [inline api/src/functions/graphql.js:7]: I don't think you need this.

File: api/src/functions/graphql.js [hunk 3]
```diff
@@ -3,11 +3,15 @@ import { createGraphQLHandler } from '@redwoodjs/graphql-server'
...
 
 import { getCurrentUser } from 'src/lib/auth.js'
 // import { db } from 'src/lib/db'
 import { logger } from 'src/lib/logger'
 
+init(config);
```
Comment (rishabhpoddar) [inline api/src/functions/graphql.js:13]: I don't think you need this.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: api/package.json (hunk 1 above)
Comment (nkshah2) [reply api/package.json]: Done

File: api/src/functions/graphql.js (hunk 2 above)
Comment (nkshah2) [reply api/src/functions/graphql.js]: Done

File: api/src/functions/graphql.js (hunk 3 above)
Comment (nkshah2) [reply api/src/functions/graphql.js]: Done

File: web/src/components/SuperTokens/SuperTokens.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { AuthProvider, useAuth } from '@redwoodjs/auth'
+import Sessions from "supertokens-auth-react/recipe/session";
+
+import ThirdPartyEmailPassword, {SignInAndUp} from "supertokens-auth-react/recipe/thirdpartyemailpassword";
+import UserTools from '../UserTools/UserTools';
+import { RedwoodApolloProvider } from '@redwoodjs/web/dist/apollo'
+
+export const SuperTokensClient = {sessions: Sessions, authRecipe: ThirdPartyEmailPassword};
```
Comment (rishabhpoddar) [inline web/src/components/SuperTokens/SuperTokens.js:8]: Now that I think of this, we don't need to get `sessions` to the AuthProvider cause that's a singleton anyway.. So just `authRecipe`

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**